src/generator.cs

```diff
@@ -1117,6 +1127,176 @@ static string FindProtocolInterface (Type type, MemberInfo pi)
 		return "I" + type.Name;
 	}
 
+	// TODO: Alex
+	public static BindAsAttribute GetBindAsAttribute (ICustomAttributeProvider cu) => GetAttribute<BindAsAttribute> (cu) ?? GetAttribute<BindAsAttribute> ((cu as MethodInfo)?.ReturnParameter);
+	public static bool HasBindAsAttribute (ICustomAttributeProvider cu) => (GetAttribute<BindAsAttribute> (cu) ?? GetAttribute<BindAsAttribute> ((cu as MethodInfo)?.ReturnParameter)) != null;
+	static bool IsSetter (MethodInfo mi) => mi.IsSpecialName && mi.Name.StartsWith ("set_", StringComparison.Ordinal);
+	bool IsSmartEnum (Type type) => (type.IsValueType && type.GetFields ().Any (f => GetAttribute<FieldAttribute> (f) != null)) || Type.GetType (type.AssemblyQualifiedName.Replace (type.Name, $"{type.Name}Extensions")) != null;
```

Comment:
I think the implementation for `IsSmartEnum` would be improve if it was a normal method body, and not an expression body (because it's quite complex).